app/src/main/java/net/gsantner/markor/frontend/MarkorDialogFactory.java

```diff
@@ -647,7 +654,18 @@ public static void showHeadlineDialog(
             };
             GsSearchOrCustomTextDialog.showMultiChoiceDialogWithSearchFilterUI(activity, dopt2);
         };
-        dopt.gravity = Gravity.TOP;
+
+        final int width = activity.getResources().getDisplayMetrics().widthPixels;
+        final int height = activity.getResources().getDisplayMetrics().heightPixels;
+        if (width > height) {
+            dopt.dialogWidth = (int) (width * 0.7);
+            dopt.dialogHeight = (int) (height * 0.95);
+        } else {
+            dopt.dialogWidth = (int) (width * 0.95);
+            dopt.dialogHeight = (int) (height * 0.8);
+        }
+        dopt.gravity = Gravity.CENTER;
```

Comment:
This looks like it would be useful for other dialogs too. Maybe abstract it out into a function ?

Reply:
It ensures that the dialog is displayed at the appropriate scale on vertical and landscape screens. I can try to extract it.